ai/nl_agent.py

```
@@ -0,0 +1,284 @@
+        """Process a natural language request and return a response."""
+        req_id = str(uuid.uuid4())
+        intent, confidence = _recognize_intent(user_input)
+        resolved_complexity = complexity or _detect_complexity(user_input)
+        resolved_fen = fen or _extract_fen(user_input)
+
+        try:
+            return await self._dispatch(
+                req_id, intent, confidence, user_input,
+                resolved_fen, resolved_complexity,
+            )
+        except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)